### PR TITLE
telco5g: create a nice HTML report from Junit files

### PR DESCRIPTION
**Pull request by sshnaidm**

### CODE REVIEW

**Comment by yuvalk** [comment]
I think you shouldnt copy the validation junit files,
unless validation failed (in which case, it can help to detect what exactly failed there)
but in all other cases, eventually validation is 100%
otherwise we wont be running any functional tests..

current report include the recurring validation tests.

**Comment by sshnaidm** [comment]
> I think you shouldnt copy the validation junit files, unless validation failed (in which case, it can help to detect what exactly failed there) but in all other cases, eventually validation is 100% otherwise we wont be running any functional tests..
> 
> current report include the recurring validation tests.

Hmm. I don't copy validation xmls, only cnftests-junit*xml . Validations are in files validation_junit.xml.  If there are validations in cnftests-junit.xml, this is a different problem.

**Comment by yuvalk** [comment]
> > I think you shouldnt copy the validation junit files, unless validation failed (in which case, it can help to detect what exactly failed there) but in all other cases, eventually validation is 100% otherwise we wont be running any functional tests..
> > current report include the recurring validation tests.
> 
> Hmm. I don't copy validation xmls, only cnftests-junit*xml . Validations are in files validation_junit.xml. If there are validations in cnftests-junit.xml, this is a different problem.

I thought so, becuase when looking at xt_u32 (as a very simple example), there are so many skips..

**Comment by sshnaidm** [comment]
Tests: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/34049/rehearse-34049-periodic-ci-openshift-release-master-nightly-4.11-e2e-telco5g/1593255059980816384/artifacts/e2e-telco5g/telco5g-cnf-tests/artifacts/test_results.html

Validations: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/34049/rehearse-34049-periodic-ci-openshift-release-master-nightly-4.11-e2e-telco5g/1593255059980816384/artifacts/e2e-telco5g/telco5g-cnf-tests/artifacts/validation_results.html

Setup tests: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/34049/rehearse-34049-periodic-ci-openshift-release-master-nightly-4.11-e2e-telco5g/1593255059980816384/artifacts/e2e-telco5g/telco5g-cnf-tests/artifacts/setup_results.html

File: ci-operator/step-registry/telco5g/cnf-tests/telco5g-cnf-tests-commands.sh [hunk 1]
```diff
@@ -338,4 +338,30 @@ echo -e "${err_msg}\n"
...
+    fi
+    xml_s="${ARTIFACT_DIR}/${feature}/setup_junit.xml"
+    if [[ -f $xml_s ]]; then
+        cp $xml_s ${ARTIFACT_DIR}/setup_junit_${feature}.xml
+    fi
+done
+
+python3 -m venv ${ARTIFACT_DIR}/myenv
+source ${ARTIFACT_DIR}/myenv/bin/activate
+git clone https://github.com/sshnaidm/html4junit.git ${ARTIFACT_DIR}/html4junit
```
Comment (yuvalk) [inline ci-operator/step-registry/telco5g/cnf-tests/telco5g-cnf-tests-commands.sh:360]: can u clone it somewhere else (outside of ${ARTIFACT_DIR} ? 
I dont think we need to keep the html4junit as an artifact..

Comment (sshnaidm) [reply ci-operator/step-registry/telco5g/cnf-tests/telco5g-cnf-tests-commands.sh:360]: I can try shared dir.

**Comment by SchSeba** [comment]
/lgtm
/approve

this looks really nice!

**Comment by yuvalk** [comment]
I know you've merged this, but I think cloning from a private repo is a bad idea and calling for future trouble

**Comment by sshnaidm** [comment]
> I know you've merged this, but I think cloning from a private repo is a bad idea and calling for future trouble

Yeah, I'd like to have it in some of openshift repos. Let's sync how can I create one.

**Comment by yuvalk** [comment]
also the clone is creating non reproducible results.
now I know, this is not a critical part of the pipeline (ie - nothing will happen if we wont get html reports)
but still.. it's a bad practice..

**Comment by sshnaidm** [comment]
> also the clone is creating non reproducible results. now I know, this is not a critical part of the pipeline (ie - nothing will happen if we wont get html reports) but still.. it's a bad practice..

What do you suggest?

**Pull request merged**